homeassistant/components/aurora_abb_powerone/config_flow.py

```diff
@@ -0,0 +1,144 @@
+"""Config flow for Aurora ABB PowerOne integration."""
+import logging
+
+from aurorapy.client import AuroraError, AuroraSerialClient
+import serial.tools.list_ports
+import voluptuous as vol
+
+from homeassistant import config_entries, core
+from homeassistant.const import CONF_ADDRESS, CONF_PORT
+
+# pylint doesn't think DOMAIN is used.  But it is used!
+# Might be this bug? https://github.com/PyCQA/pylint/issues/3445
+# pylint: disable=unused-import
+from .const import (
+    ATTR_FIRMWARE,
+    ATTR_MODEL,
+    ATTR_SERIAL_NUMBER,
+    DEFAULT_ADDRESS,
+    DEFAULT_INTEGRATION_TITLE,
+    DOMAIN,
+    MAX_ADDRESS,
+    MIN_ADDRESS,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def validate_and_connect(hass: core.HomeAssistant, data):
+    """Validate the user input allows us to connect.
+
+    Data has the keys from DATA_SCHEMA with values provided by the user.
+    """
+    comport = data[CONF_PORT]
+    address = data[CONF_ADDRESS]
+    _LOGGER.debug("Intitialising com port=%s", comport)
+    ret = {}
+    ret["title"] = DEFAULT_INTEGRATION_TITLE
+    try:
+        client = AuroraSerialClient(address, comport, parity="N", timeout=1)
+        client.connect()
+        ret[ATTR_SERIAL_NUMBER] = client.serial_number()
+        ret[ATTR_MODEL] = f"{client.version()} ({client.pn()})"
+        ret[ATTR_FIRMWARE] = client.firmware(1)
+        _LOGGER.info("Returning device info=%s", ret)
+    except AuroraError as err:
+        _LOGGER.warning("Could not connect to device=%s", comport)
+        raise err
+    finally:
+        if client.serline.isOpen():
+            client.close()
+
+    # Return info we want to store in the config entry.
+    return ret
+
+
+class AuroraABBConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Aurora ABB PowerOne."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+    _comportslist = None
+    _defaultcomport = None
```

Comment:
These should be defined in `__init__` since they probably aren't intended to be class variables

Reply:
Done.